NetworkingComparison/RxSwift/RxSwiftAPI.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  RxSwiftAPI.swift
+//  NetworkingComparison
+//
+//  Created by Ian Luo on 11/18/19.
+//  Copyright © 2019 Lokae. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+import RxCocoa
+
+class RxSwiftAPI {
+    static let shared = RxSwiftAPI()
+
+    enum RxSwiftError: Error {
+        case badUrl, invalidKey, cityNotFound, serverFailure
+    }
+
+    private let session = URLSession.shared
+
+    private enum SessionStyle {
+        case observableRequest, standardRequest, manualResponse
+    }
+
+    private var sessionStyle: SessionStyle = .observableRequest
+
+    private init() { }
+
+    func getForecasts() -> Observable<[CodableForecast]> {
+        var data: Observable<Data>
+        switch sessionStyle {
+        case .observableRequest: data = fetchWithStandardRequest()
+        case .standardRequest: data = fetchWithObservableRequest()
+        case .manualResponse: data = fetchWithManualResponse()
+        }
+        return data.map {
+            let decoder = JSONDecoder()
```

Comment:
Would it be preferable to initialize and setup this decoder outside (above) of this closure?

It's a tradeoff of memory vs performance, but I think in this case it may be preferable to only instantiate one decoder.